core/src/main/java/org/stellar/anchor/util/Sep1Helper.java

```diff
@@ -1,24 +1,43 @@
 package org.stellar.anchor.util;
 
+import static org.stellar.anchor.util.Log.*;
+
 import com.moandjiezana.toml.Toml;
 import java.io.IOException;
 import java.net.URL;
+import org.stellar.anchor.api.exception.InvalidConfigException;
 
 public class Sep1Helper {
   public static TomlContent readToml(String url) throws IOException {
     return new TomlContent(new URL(url));
   }
 
-  public static TomlContent parse(String tomlString) {
-    return new TomlContent(tomlString);
+  public static TomlContent parse(String tomlString) throws InvalidConfigException {
+    try {
+      return new TomlContent(tomlString);
+    } catch (Exception e) {
+      // Obfuscate the message and rethrow
+      String obfuscatedMessage = "Failed to parse TOML content. Invalid Config.";
+      debugF(e.toString()); // Log the exception
+      throw new InvalidConfigException(
+          obfuscatedMessage); // Preserve the original exception as the cause
+    }
   }
 
   public static class TomlContent {
     private final Toml toml;
 
     TomlContent(URL url) throws IOException {
-      String tomlValue = NetUtil.fetch(url.toString());
-      toml = new Toml().read(tomlValue);
```

Comment:
It looks like we are handling the exceptions in both in the `TomlContent` constructor and the `Sep1Helper` methods. Do you think it would be cleaner to move all the exception handling into `Sep1Helper` and removing `TomlContent(URL url)`?

It would look something like this:
```java
public class Sep1Helper {
    public static TomlContent readToml(String url) throws IOException {
        try {
            String tomlValue = NetUtil.fetch(url);
            return TomlContent.parse(tomlValue);
        } catch (IOException e) {
            debugF("Failed to fetch toml file from {} with message: {}", url, e.toString());
            throw new IOException("Failed to fetch toml file from " + url);
        }
    }

    public static TomlContent parse(String tomlString) throws InvalidConfigException {
        try {
            return new TomlContent(tomlString);
        } catch (Exception e) {
            debugF("Failed to parse toml file with message: {}", e.toString());
            throw new ("Failed to parse TOML content. Invalid Config.");
        }
    }
}
```

Reply:
nice one